### PR TITLE
feat(perf-issues): Add performance issue support to `EventGroupingInfoEndpoint`

### PR DESCRIPTION
**Pull request by gggritso**
We want to show grouping information for performance issue events, but performance issues use a _completely_ different grouping system from errors. Instead of using the existing variants and configs, performance issues are fingerprinted by the performance issue detector, and they store the detection output (AKA "evidence") in nodestore.

Instead of doing a frontend-only hack, this PR takes some steps to bridge the gap. Adds a new `PerformanceProblemVariant` class that delegates its hash to an `EventPerformanceProblem`. This variant returns the detector information that's used for fingerprinting (the op, and the span hashes). When fetching grouping information for a transaction event, manually sets up the variants by fetching the corresponding `EventPerformanceProblem`s.

This is the UI this PR enables (ignore the text changes):

Screen Shot 2022-09-20 at 5 01 58 PM

Partially closes PERF-1758 and ISP-33

### CODE REVIEW

File: tests/sentry/api/endpoints/test_event_grouping_info.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+from sentry.utils import json
+from sentry.utils.performance_issues.performance_detection import (
+    EventPerformanceProblem,
+    PerformanceProblem,
+)
+from sentry.utils.samples import load_data
+
+
+@region_silo_test
+class ProjectOwnershipEndpointTestCase(APITestCase):
```
Comment (barkbarkimashark) [inline tests/sentry/api/endpoints/test_event_grouping_info.py:16]: Should this be named `EventGroupingInfoEndpointTestCase` or something like that?

**Review by barkbarkimashark — APPROVED** [review]
lgtm, besides the testname!

File: tests/sentry/api/endpoints/test_event_grouping_info.py (hunk 1 above)
Comment (gggritso) [reply tests/sentry/api/endpoints/test_event_grouping_info.py]: Whoops, great catch, thanks!

**Review by k-fish — APPROVED** [review]
Not sure whether variant changes belong in the model or serialize layers, but 👍 for now, we can always quickly fix this later.

File: src/sentry/api/endpoints/event_grouping_info.py [hunk 2]
```diff
@@ -39,9 +42,26 @@ def get(self, request: Request, project, event_id) -> Response:
...
+                problems = EventPerformanceProblem.fetch_multi([(event, h) for h in hashes.hashes])
+
+                # Create a variant for every problem associated with the event
+                # TODO: Generate more unique keys, in case this event has more than
+                # one problem of a given type
+                variants = {
+                    problem.problem.type.name: PerformanceProblemVariant(problem)
+                    for problem in problems
+                    if problem
+                }
```
Comment (wedamija) [inline src/sentry/api/endpoints/event_grouping_info.py:59]: This logic probably should be moved to `get_grouping_variants` at some point

Comment (gggritso) [reply src/sentry/api/endpoints/event_grouping_info.py:59]: Makes sense!

**Pull request merged**